server/sonar-server/src/main/java/org/sonar/server/measure/ws/ComponentTreeDataLoader.java

```diff
@@ -110,17 +104,16 @@ ComponentTreeData load(ComponentTreeWsRequest wsRequest) {
       }
       Long developerId = searchDeveloperId(dbSession, wsRequest);
 
-      ComponentTreeQuery dbQuery = toComponentTreeQuery(wsRequest, baseComponent);
-      ComponentDtosAndTotal componentDtosAndTotal = searchComponents(dbSession, dbQuery, wsRequest);
-      List<ComponentDto> components = componentDtosAndTotal.componentDtos;
+      ComponentTreeQuery componentTreeQuery = toComponentTreeQuery(wsRequest, baseComponent);
+      List<ComponentDto> components = searchComponents(dbSession, componentTreeQuery);
       List<MetricDto> metrics = searchMetrics(dbSession, wsRequest);
       List<WsMeasures.Period> periods = snapshotToWsPeriods(baseSnapshot.get());
-      Table<String, MetricDto, MeasureDto> measuresByComponentUuidAndMetric = searchMeasuresByComponentUuidAndMetric(dbSession, baseComponent, components, metrics,
-        periods, developerId);
+      Table<String, MetricDto, MeasureDto> measuresByComponentUuidAndMetric = searchMeasuresByComponentUuidAndMetric(dbSession, baseComponent, componentTreeQuery, components,
+        metrics, periods, developerId);
 
       components = filterComponents(components, measuresByComponentUuidAndMetric, metrics, wsRequest);
+      int componentCount = components.size();
```

Comment:
this statement could be moved just before paginateComponents(), so that all stuff around pagination is at the same place.